tests/test_permissions_cli.py

```diff
@@ -0,0 +1,348 @@
+        },
+        "deposit-update": {
+            "roles": [],
+            "users": ["info@inveniosoftware.org"]
+        }
+    })
+
+
+def test_permissions_add_no_pid_given(cli_run):
+    res = cli_run("permissions add -u cms@inveniosoftware.com -r read")
+
+    assert res.exit_code == 2
+    assert "Error: Missing option '--pid' / '-p'." in res.output
+
+
+def test_permissions_add_no_user_given(cli_run):
+    res = cli_run("permissions add -p some-pid -r read")
+
+    assert res.exit_code == 2
+    assert "Error: You need to specify --user or --egroup." in res.output
+
+
+def test_permissions_add_no_rights_given(cli_run):
+    res = cli_run("permissions add -p some-pid -u cms@inveniosoftware.com")
+
+    assert res.exit_code == 2
+    assert "Error: Missing option '--rights' / '-r'." in res.output
+
+
+def test_permissions_add_wrong_rights_given(cli_run):
+    res = cli_run(
+        "permissions remove -p some-pid -u cms@inveniosoftware.com -r RAND")
+
+    assert res.exit_code == 2
+    assert "Error: Invalid value for '--rights' / '-r': " \
+           "invalid choice: RAND. (choose from read, update, admin)" in res.output
+
+
+@responses.activate
+def test_permissions_add_pid_not_exists(cli_run):
+    responses.add(
+        responses.POST,
+        'https://analysispreservation-dev.cern.ch/api/deposits/some-pid/actions/permissions',
+        json={
+            'status': 404,
+            'message': 'PID does not exist.'
+        },
+        status=404)
+
+    res = cli_run(
+        "permissions add -p some-pid -u cms@inveniosoftware.com -r read")
+
+    assert res.exit_code == 1
+    assert res.stripped_output == 'PID does not exist.'
+
+
+@responses.activate
+def test_permissions_add_user_not_exists(cli_run):
+    responses.add(
+        responses.POST,
+        'https://analysispreservation-dev.cern.ch/api/deposits/some-pid/actions/permissions',
+        json={
+            'status': 400,
+            'message': 'User with this mail does not exist in LDAP.'
+        },
+        status=400)
+
+    res = cli_run(
+        "permissions add -p some-pid -u cms@inveniosoftware.com -r read")
+
+    assert res.exit_code == 1
+    assert res.stripped_output == 'User with this mail does not exist in LDAP.'
+
+
+@responses.activate
+def test_permissions_add_no_access(cli_run):
+    responses.add(
+        responses.POST,
+        'https://analysispreservation-dev.cern.ch/api/deposits/some-pid/actions/permissions',
+        json={
+            'message': "You don't have the permission to access the requested resource. "
+            "It is either read-protected or not readable by the server.",
+            'status': 403
+        },
+        status=403)
+
+    res = cli_run(
+        "permissions add -p some-pid -u cms@inveniosoftware.com -r read")
+
+    assert res.exit_code == 1
+    assert res.stripped_output == "You don't have sufficient permissions."
+
+
+# REMOVE
+@responses.activate
+def test_permissions_remove(cli_run):
```

Comment:
same like above, checks for request headers and body, plus add tests for removing egroup permissions